src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/GetObjectForNativeVariantTests.cs

```diff
@@ -51,133 +56,1059 @@ internal struct TypeUnion
+
+            // VT_R4 => float.
+            yield return new object[]
+            {
+                CreateVariant(VT_R4, new UnionTypes { _r4 = 10 }),
+                (float)10
+            };
+
+            yield return new object[]
+            {
+                CreateVariant(VT_R4, new UnionTypes { _r4 = 0 }),
+                (float)0
+            };
+
+            yield return new object[]
+            {
+                CreateVariant(VT_R4, new UnionTypes { _r4 = -10 }),
+                (float)(-10)
+            };
+
+            yield return new object[]
+            {
+                CreateVariant(VT_R4, new UnionTypes { _r4 = float.PositiveInfinity }),
+                float.PositiveInfinity
+            };
+
+            yield return new object[]
+            {
+                CreateVariant(VT_R4, new UnionTypes { _r4 = float.NegativeInfinity }),
+                float.NegativeInfinity
+            };
+
+            yield return new object[]
+            {
+                CreateVariant(VT_R4, new UnionTypes { _r4 = float.NaN }),
+                float.NaN
+            };
+
+            // VT_R8 => double.
+            yield return new object[]
+            {
+                CreateVariant(VT_R8, new UnionTypes { _r8 = 10 }),
+                (double)10
+            };
 
-        private static void DeleteVariant(IntPtr pVariant)
+            yield return new object[]
+            {
+                CreateVariant(VT_R8, new UnionTypes { _r8 = 0 }),
+                (double)0
+            };
+
+            yield return new object[]
+            {
+                CreateVariant(VT_R8, new UnionTypes { _r8 = -10 }),
+                (double)(-10)
+            };
+
+            yield return new object[]
+            {
+                CreateVariant(VT_R8, new UnionTypes { _r8 = double.PositiveInfinity }),
+                double.PositiveInfinity
+            };
+
+            yield return new object[]
+            {
+                CreateVariant(VT_R8, new UnionTypes { _r8 = double.NegativeInfinity }),
+                double.NegativeInfinity
+            };
+
+            yield return new object[]
+            {
+                CreateVariant(VT_R8, new UnionTypes { _r8 = double.NaN }),
+                double.NaN
+            };
+
+            // VT_CY => decimal.
+            yield return new object[]
+            {
+                CreateVariant(VT_CY, new UnionTypes { _cy = 200 }),
+                0.02m
+            };
+
+            yield return new object[]
+            {
+                CreateVariant(VT_CY, new UnionTypes { _cy = 0 }),
+                0m
+            };
+
+            yield return new object[]
+            {
+                CreateVariant(VT_CY, new UnionTypes { _cy = -200 }),
+                -0.02m
+            };
+
+            // VT_DATE => DateTime.
+            yield return new object[]
+            {
+                CreateVariant(VT_DATE, new UnionTypes { _date = 2958465.99999999 }),
```

Comment:
Any way you can make a local and have the raw value computed instead of hard coding `2958465...`

Reply:
Fixed